pip/req/req_set.py

```diff
@@ -288,33 +296,247 @@ def uninstall(self, auto_confirm=False):
             req.uninstall(auto_confirm=auto_confirm)
             req.commit_uninstall()
 
-    def _walk_req_to_install(self, handler):
-        """Call handler for all pending reqs.
-
-        :param handler: Handle a single requirement. Should take a requirement
-            to install. Can optionally return an iterable of additional
-            InstallRequirements to cover.
-        """
-        # The list() here is to avoid potential mutate-while-iterating bugs.
-        discovered_reqs = []
-        reqs = itertools.chain(
-            list(self.unnamed_requirements), list(self.requirements.values()),
-            discovered_reqs)
-        for req_to_install in reqs:
-            more_reqs = handler(req_to_install)
-            if more_reqs:
-                discovered_reqs.extend(more_reqs)
-
     def prepare_files(self, finder):
         """
         Prepare process. Create temp directories, download and/or unpack files.
+
+        RESOLVE
         """
         # make the wheelhouse
         if self.wheel_download_dir:
             ensure_dir(self.wheel_download_dir)
+        # Allow for a decent # of reqs
+        # Each dep in the graph is a frame.
+        sys.setrecursionlimit(5000)
```

Comment:
Should this just be dynamically computed (since we should know the height of the tree)? Perhaps something for later as this gets optmized more...